### DISCUSSION TITLE
Cannot decrypt ReadStream from file

### DISCUSSION THREAD

**Discussion by EricMarcantonio** [openpgpjs/openpgpjs#1305] [Q&A] [answered]
Hello.

I am trying to symmetrically decrypt a file using a 
```js 
fs.ReadStream
```

Here is my encryption and decryption:
```js
async function encryptFileSym(fileBuffer: fs.ReadStream, secret: string) {
    const message = await openpgp.createMessage({ text: fileBuffer });
    return await openpgp.encrypt({
        message,
        passwords: secret,
        armor: false, 
    });
}
async function decryptFileSym(fileBuffer: fs.ReadStream, secret: string) {
    const message = await openpgp.readMessage({ armoredMessage: fileBuffer });
    return await openpgp.decrypt({
        message,
        passwords: secret,
    });
}

async function encryptFileAndCreateReplacement(filename: string, secret: string) {
    const newFileName = filename + ".enc";
    const matchingKeyName = filename + ".key.enc";
    const oldFileBuff = fs.createReadStream(filename);
    const encryptedData = await encryptFileSym(oldFileBuff, secret);
    const newFileStream = fs.createWriteStream(newFileName);
    encryptedData.pipe(newFileStream);
}

async function decryptFileAndCreateReplacement(filename: string, secret: string) {
    const newFileName = filename.replace(".enc", "plswork");
    //delete the old file
    //delete the encrypted secret file
    const oldFileBuff = fs.createReadStream(filename);
    const decryptedData = await decryptFileSym(oldFileBuff, secret);
    const newFileStream = fs.createWriteStream(newFileName);
    decryptedData.data.pipe(newFileStream);
}
openpgp.config.allowUnauthenticatedStream = true;

(async function () {
    await encryptFileAndCreateReplacement("secret", "helloworld");
    await decryptFileAndCreateReplacement("secret.enc", "helloworld");
})();
```
It throws a `UnhandledPromiseRejectionWarning: Error: Misformed armored text`.

In essence, I am writing the encrypted to a file, and then trying to read that file and decrypt. **Is this a bug, or am I missing something?** I have also changed `{ text: fileBuffer }` to `{ binary: fileBuffer }` 

Thank you for your continued support,
Eric

**Comment by twiss** [comment]
Hey :wave: Probably you have to pass `'utf8'` as the second parameter to `createReadStream` when decrypting an armored file.

**Reply by EricMarcantonio** [reply]
Thank you for your reply.

I have changed the encoding of the `createWriteStream` to utf-8 and the reading of the encrypted file to utf-8. This produces no errors. 🥳 

However, this only works using text-based data. When trying to encrypt a binary file, it corrupts the binary.

Do you have any idea why this would occur?

Here is [v2](https://github.com/EricMarcantonio/openpgp-lambda-crypto/blob/master/src/index.ts) of the code that is failing.

Thank you,
Eric

**Comment by EricMarcantonio** [comment, marked as answer]
Hey! I have found the solution.

When decrypting `binary` files, `{ format: "binary" }` needs to be added or you will lose carriage returns, etc.

Thank you for your support, I hope this thread helps someone who might be looking! 🥳

**Comment by Tzvetelin88** [comment]
Hello, is it possible to use this code to encrypt/decrypt ZIP file? As I have tested it, but without success :(, thanks!

**Reply by EricMarcantonio** [reply]
Yes please see above. The code should work if you add in my suggestion (format: binary)

**Reply by Tzvetelin88** [reply]
Yep, I have try this. 
For encryption I'm getting this:
"
-----BEGIN PGP MESSAGE-----

wy4ECQMI1Jr0JFoxig3gbrsv92Cogyts2PBYLsmV4PLNNF7HJCXsMoodlKsO
Rsy60jEB9q3Np4lQx/atT240bOzNIbrNC7FXlSknkokUPsicgka6SPXNeGLi
YJ6CxPzHf6e5
=U0aU
-----END PGP MESSAGE-----
"

but for decryption I'm getting this: 'UnhandledPromiseRejectionWarning: Error: Misformed armored text' using the code from above, just changing this:
"const message = await openpgp.createMessage({ text: fileBuffer, type: 'binary' });" 

Also I'm using version: ^5.0.0-3.
I notice that there are a lot of changes compared to 4.10.10 for example

Thanks

**Reply by EricMarcantonio** [reply]
Hey maybe take a look here?

https://github.com/EricMarcantonio/openpgp-lambda-crypto/blob/master/src/index.ts

This is what I used openpgp for, maybe this can help? If you send the link to your repo I can also take a look.

Best,
Eric

**Reply by Tzvetelin88** [reply]
Thanks so much Eric, please find repo here: https://github.com/Tzvetelin88/pgp-zip . I have one simple test.zip containing one text file with data. test-copy.zip is just for backup of the original zip. Encryption is creating new test-enc.zip and decryption creates test-dec.zip

**Reply by Tzvetelin88** [reply]
I have fix it. I needed to change a lot of stuff same code in https://github.com/Tzvetelin88/pgp-zip
